addons/core/functions/fnc_parseFrequenciesInput.sqf

```diff
@@ -0,0 +1,62 @@
+#include "script_component.hpp"
+
+/*
+    Name: TFAR_fnc_parseFrequenciesInput
+
+    Author(s):
+        Dorbedo
+
+    Description:
+        parses a frequencies array and fills it up with default values
+        does a valuecheck and ignores wrong values
+
+    Parameters:
+        STRING: value string
+        SCALAR: amount of channels
+        SCALAR: min freq
+        SCALAR: max freq
+        SCALAR: round power
+
+    Returns:
+        NOTHING
+
+    Example:
+        ["[50.16549, 51122, 52, 53, 4, 5, 56, 57, 58, 59, ""asd"", asd, "88"]", 8, 87, 40] call TFAR_fnc_parseFrequenciesInput;
+*/
+
+params [
+    ["_valueString","",[""]],
+    ["_maxChannels",TFAR_MAX_CHANNELS,[TFAR_MAX_CHANNELS]],
+    ["_maxFreq",TFAR_MAX_SW_FREQ,[TFAR_MAX_SW_FREQ]],
+    ["_minFreq",TFAR_MIN_SW_FREQ,[TFAR_MIN_SW_FREQ]],
+    ["_roundPower",TFAR_FREQ_ROUND_POWER,[TFAR_FREQ_ROUND_POWER]]
+];
+
+// add brackets if they are missing
+if !(((_valueString find "[") isEqualTo 0) && {(_valueString find "]") isEqualTo ((count _valueString) - 1)}) then {
+    _valueString = format["[%1]",_valueString];
+};
+
+private _parsedValue = (parseSimpleArray _valueString) apply {
+    if (IS_STRING(_x)) then {parseNumber _x} else {_x};
+};
+
+_parsedValue = _parsedValue select {
+    IS_NUMBER(_x) &&
```

Comment:
This will just remove invalid thingies right? We might want to insert a random value. Instead of moving the ones after the invalid one to the front.
If user has 1,2,3 set and 4 invalid and 5,6,7 set.
then their 5 frequency will all of the sudden land on 4.

I don't care though. Just a thing you could do if you feel like it.

Reply:
In my opinion, the mission-designer should just fix it.

The target of this function was just to prevent invalid stuff.